src/RichMentionsProvider.tsx

```diff
@@ -58,15 +59,27 @@ export function RichMentionsProvider<T = object>({
   };
 
   const onBeforeChanges: TMentionContext['onBeforeChanges'] = event => {
-    const selection = document.getSelection();
+    let selection = document.getSelection();
+
     if (!selection || !selection.anchorNode) {
       return;
     }
 
+    // If there is text selection, delete it.
+    // We need to do it manually because of the preventDefault() :'(
+    // Update 'text' variable as the content could be updated
+    if (deleteSelection(selection)) {
+      selection = document.getSelection();
+      if (!selection || !selection.anchorNode) {
+        return;
+      }
```

Comment:
Selection is not voided after deletion ?
Is there a case where this is false ?

Reply:
After deletion, the browser change the selection : 
- The selection could just disappear for some reason (blur).
- The selection could be updated (offset, anchorNode changes).

Let say you have: `text <span>@|def</span>` where "|" is the cursor.
You type "backspace", deleting the "@", then the fragment is deleted.

In this situation, the browser will change your selection after deletion:
- anchorNode : SpanElement -> Text
- anchorOffset: 1 -> 5